### PR TITLE
config: Add Lock View's config to config example and test

### PR DESCRIPTION
**Pull request by MyonKeminta**
### What problem does this PR solve?

Problem Summary: [Lock View](https://github.com/pingcap/tidb/issues/24199)'s config `pessimistic-txn.deadlock-history-capacity` is not present in config example and the tests. cc https://github.com/pingcap/tidb/issues/25332

### What is changed and how it works?

What's Changed: Added the config `pessimistic-txn.deadlock-history-capacity`  to the config example file and TestConfig.

### Related changes

- Need to cherry-pick to the release branch
  - release-5.1

### Check List 

Tests 

- Unit test

### Release note 

- No release note

### CODE REVIEW

**Comment by MyonKeminta** [comment]
cc @longfangsong  Please remember to update the tests and config example when adding new configs, unless it's supposed to be undocumented.

**Review by cfzjywxk — APPROVED** [review]
/lgtm

**Review by longfangsong — APPROVED** [review]

File: config/config.go [hunk 1]
```diff
@@ -495,6 +495,7 @@ type Binlog struct {
 type PessimisticTxn struct {
 	// The max count of retry for a single statement in a pessimistic transaction.
 	MaxRetryCount           uint `toml:"max-retry-count" json:"max-retry-count"`
```
Comment (longfangsong) [inline config/config.go:497]:
```suggestion
	MaxRetryCount uint `toml:"max-retry-count" json:"max-retry-count"`
```

Comment (MyonKeminta) [reply config/config.go:497]: Ah, thanks

**Pull request merged**